diff --git a/zonos/utils.py b/zonos/utils.py
index 6e833f3..b8d1530 100644
--- a/Zonos/zonos/utils.py
+++ b/Zonos/zonos/utils.py
@@ -29,11 +29,17 @@ def pad_weight_(w: nn.Embedding | nn.Linear, multiple: int):
 
 def get_device() -> torch.device:
     if torch.cuda.is_available():
-        return torch.device(torch.cuda.current_device())
+        device = torch.device(torch.cuda.current_device())
     # MPS breaks for whatever reason. Uncomment when it's working.
-    # if torch.mps.is_available():
+    # elif torch.mps.is_available():
     #     return torch.device("mps")
-    return torch.device("cpu")
+    elif torch.xpu.is_available():
+        device = torch.device("xpu")
+    else:
+        device = torch.device("cpu")
+
+    print(f"Using {device}")
+    return device
 
 
 DEFAULT_DEVICE = get_device()
